### DISCUSSION TITLE
Why do we even consider 'sys.modules' in 'swift.swift._add_to_path'?

### DISCUSSION THREAD

**Discussion by BECATRUE** [snu-quiqcl/qiwis#111] [Q&A] [answered]
I have a question about `_add_to_path()` function in `swift.swift` module.

The function is as below:
```python
@contextmanager
def _add_to_path(path: str):
    """Adds a path temporarily.

    Using a 'with' statement, you can import a module without changing sys.path.

    Args:
        path: A desired path to be added. 
    """
    old_modules = sys.modules
    sys.modules = old_modules.copy()
    old_path = sys.path
    sys.path = old_path.copy()
    sys.path.insert(0, path)
    try:
        yield
    finally:
        sys.path = old_path
        sys.modules = old_modules
```

In the function, we change only `sys.path`, but why do we even care about `sys.modules`?

I think it is not neccessary to consider `sys.modules`.

**Comment by BECATRUE** [comment]
In the [docs](https://docs.python.org/3/library/sys.html#sys.modules) about `sys.modules`, it is described as below.
>This is a dictionary that maps module names to modules which have already been loaded. This can be manipulated to force reloading of modules and other tricks. However, replacing the dictionary will not necessarily work as expected and deleting essential items from the dictionary may cause Python to fail. If you want to iterate over this global dictionary always use sys.modules.copy() or tuple(sys.modules) to avoid exceptions as its size may change during iteration as a side effect of code or activity in other threads.

So, I think even if you import a module later, it will not change `sys.modules`.

**Reply by kangz12345** [reply]
Oh, great search! That scares me a lot.
Please make an issue of this and implement it.

**Reply by BECATRUE** [reply]
Then, do I just have to remove the part about `sys.modules`?

**Reply by kangz12345** [reply]
Yes please!

**Comment by BECATRUE** [comment, marked as answer]
What do you think of it, @kangz12345?
